File: Map/Z_InterpretorWCDL/src/expression/ConstExpression.java
package expression;

import domain.IDictionary;
import domain.IHeap;

public class ConstExpression implements IExpression
{

	private int constant;
	
	public ConstExpression(int value)
	{
		constant=value;
	}
	
	@Override
	public int eval(IDictionary<String, Integer> symbolTable, IHeap<Integer,Integer> heap) 
	{
		return constant;
	}
	
	@Override 
	public String toString()
	{
		return ""+constant;
	}

}

File: Map/Map/Map/expression/ConstExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Map.domain;

namespace Map.expression
{
    class ConstExpression : IExpression
    {
        private int constant;

        public ConstExpression(int val)
        {
            constant = val;
        }

        public int eval(IDict<string, int> symbolTable)
        {
            return constant;
        }

        public override String ToString()
        {
            return constant + "";
        }
    }
}
